app/src/main/kotlin/com/intive/picover/parties/viewmodel/PartiesViewModel.kt

```diff
@@ -3,23 +3,38 @@ package com.intive.picover.parties.viewmodel
 import androidx.compose.runtime.MutableState
 import androidx.compose.runtime.mutableStateOf
 import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
 import com.intive.picover.parties.repository.MockedPartiesRepository
 import com.intive.picover.parties.state.PartiesState
+import com.intive.picover.parties.state.PartiesState.Error
+import com.intive.picover.parties.state.PartiesState.Loaded
+import com.intive.picover.parties.state.PartiesState.Loading
 import dagger.hilt.android.lifecycle.HiltViewModel
 import javax.inject.Inject
+import kotlinx.coroutines.launch
 
 @HiltViewModel
 class PartiesViewModel @Inject constructor(
 	private val partiesRepository: MockedPartiesRepository,
 ) : ViewModel() {
 
-	val parties: MutableState<PartiesState> = mutableStateOf(PartiesState.Loading)
+	val parties: MutableState<PartiesState> = mutableStateOf(Loading)
 
 	init {
-		loadParties()
+		viewModelScope.launch {
+			loadParties()
+		}
 	}
 
-	private fun loadParties() {
-		parties.value = PartiesState.Loaded(partiesRepository.getParties())
+	private suspend fun loadParties() {
+		runCatching {
```

Comment:
I think that `runCatching` is wrongly used here, because we are only checking exception of the `partiesRepository.parties()` but we should check exceptions which could be thrown during Flow collection, so it should be rather something like that:
```
		runCatching {
			partiesRepository.parties().collect {
				parties.value = Loaded(it)
			}
		}.onFailure {
			parties.value = Error
		}
```

Reply:
...make sense :D i am wondering about the types of exception that could be thrown by the collection itself, but i think this is something to dive deeper in :D

Reply:
I think that all of the exceptions will be thrown during the collection for example when there is no internet connection, without the `collect` there will be no exception, because the call will not start :D

Reply:
You are right -> used catch in this way and it works also properly :) 

What is interesting, tests stop working for checking error case, i had to use:

```
every { repository.partyById(partyId) } returns flow<PartyRemote> { throw Exception() }
```
instead of:

```
every { repository.partyById(partyId) } throws Exception()
```

...which convinced me, that thanks to this, i can use parameterised tests here, and we can do it for all cases in VM's when we want to check states, as it can be written in this way : 

```
emptyFlow<PartyRemote>() to Loading,
flowOf(partyRemote) to Loaded(party),
flow<PartyRemote> { throw Exception() } to Error,
```
But one doubt come into my mind, in which way we want to do it (as we are using data fetching in init block, and we have to init viewmodel for every single test): 
- In the way i've implemented in this PR for viewmodels, that we have **ONE** test for every value, with forAll usage and initing viewmodel for every iteration, using nice features of kotest forAll usage, especially in failure info printing
- In the way that is for example implmented in `MainStateTest`, that we have **MANY** tests, with isolation mode per method, without usage of for all, but we don't have to init viewmodel in every single test of parameterised, without forAll usage

You can check this comment @kmaslowiec @BorysChajdas @MarekMacko @bartek977 @michal-kucznerowicz @Nataniel-Antosik

Reply:
I would go with one test ;)

Reply:
Same I would go with one test 😄